### PR TITLE
Making `\` work with SparseMatrix.CHOLMOD.Factor and SubArrays

### PR DESCRIPTION
**Pull request by p-zubieta**
This is needed by #16294 (see the discussion over there).

CC @andreasnoack

### CODE REVIEW

**Comment by tkelman** [comment]
Can we add tests of these on strided inputs? I think a SubArray would suffice?

File: base/sparse/cholmod.jl [hunk 1]
```diff
@@ -1464,7 +1464,7 @@ Ac_ldiv_B(L::FactorComponent, B) = ctranspose(L)\B
 
 (\){T}(L::Factor{T}, B::Dense{T}) = solve(CHOLMOD_A, L, B)
 (\)(L::Factor{Float64}, B::VecOrMat{Complex{Float64}}) = L\real(B) + L\imag(B)
-(\)(L::Factor, b::Vector) = Vector(L\convert(Dense{eltype(L)}, b))
+(\)(L::Factor, b::StridedVector) = Vector(L\convert(Dense{eltype(L)}, b))
 (\)(L::Factor, B::Matrix) = Matrix(L\convert(Dense{eltype(L)}, B))
```
Comment (andreasnoack) [inline base/sparse/cholmod.jl]: This one should also be changed to `StridedMatrix` although it's not needed for `eigs`.

**Comment by pabloferz** [comment]
Tests added, comments addressed and green lights. Is this ready to go?

**Comment by andreasnoack** [comment]
Thanks for fixing this.

**Comment by pabloferz** [comment]
No problem.

**Pull request merged**